server/routes/webhooks/stripe-charge-failed.js

```diff
@@ -0,0 +1,106 @@
+const boom = require('boom');
+const stripe = require('../stripe');
+const basket = require('../../lib/basket-queue');
+
+// values that never change between webhook requests
+const signatureHeader = 'stripe-signature';
+const event_type = 'charge.failed';
+const frequency = 'monthly';
+const recurring = true;
+const service = 'stripe';
+
+// Error/Success message constants
+const forbiddenError = 'An error occurred while verifying the webhook signing secret';
+const incorrectEventError = 'This hook only processes charge failed events';
+const expandChargeError = 'An error occurred while expanding the charge';
+const retrieveSubscriptionErrorMsg = 'An error occurred while fetching the subscription';
+const notRecurringCharge = 'This hook only processes recurring charges that fail';
+const notPartOfSubscription = 'This charge is not part of a subscription';
+const successMessage = 'charge failed event processed';
+
+let stripeChargeFailed = (request, reply) => {
+  // Validate the Webhook signature
+  let endpointSecret = process.env.STRIPE_WEBHOOK_SIGNATURE_CHARGE_FAILED;
+  let signature = request.headers[signatureHeader];
+
+  let event = stripe.constructEvent(request.payload, signature, endpointSecret);
+
+  // If there's no event object, the signature verification failed
+  if (!event) {
+    return reply(boom.forbidden(forbiddenError));
+  }
+
+  // Only process charge.failed webhook events
+  if (event.type !== event_type) {
+    return reply(incorrectEventError);
+  }
+
+  let {charge} = event.data.object;
```

Comment:
why do we destructure `event.data.object`? Is there a `charge` property on `event.data.object`?

Reply:
Good catch